packages/core/src/odata/common/uri-conversion/get-filter.ts

```diff
@@ -130,6 +135,21 @@ export function createGetFilter(uriConverter: UriConverter) {
         uriConverter.convertToUriFormat(filter.value, filter.edmType!)
       ].join(' ');
     }
+
+    if (isFilterLambdaExpression(filter)) {
+      const alias = `a${lambdaExpressionLevel}`;
+      const filterExp = getODataFilterExpression(
+        filter.filters,
+        [alias],
+        targetEntityConstructor,
+        lambdaExpressionLevel + 1
+      );
+      return `${parentFieldNames.join('/')}/${
+        filter.lambdaOperator
+      }(${alias}:${removeBrackets(filterExp)})`;
+    }
+
+    throw new Error(`Filter: ${filter} is not valid.`);
```

Comment:
Discussed, this should never be reached, since all the types of the `Filterable` are considered.
With the error message, the function can return `string` instead of `string|undefined`, and all the called can stop considering the null check.